### PR TITLE
Premises characteristics can be saved

### PR DESCRIPTION
**Pull request by bryangaledxw**
# Changes in this PR

* A user can enter characteristics for a premises
* A user can see characteristics for a premises

### Before
localhost_3000_properties_d84c6ad8-7334-4fcd-8191-ec6784af0844 (1)
localhost_3000_properties_new (2)

### After
localhost_3000_properties_d84c6ad8-7334-4fcd-8191-ec6784af0844
localhost_3000_properties_new (1)

### CODE REVIEW

**Comment by bryangaledxw** [comment]
[Premises characteristics can be saved](https://trello.com/c/cOmYoJ8T/476-premises-characteristics-can-be-saved)

File: server/utils/characteristicUtils.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Characteristic } from '../@types/shared'
+
+export const prepareCharacteristics = (
```
Comment (richpjames) [inline server/utils/characteristicUtils.ts]: 'prepare' is a bit vague as a name to me but I can't suggest anything better and the function body is very clear so I think it's fine for now.

File: e2e/tests/stepDefinitions/manage.ts [hunk 2]
```diff
@@ -28,10 +29,17 @@ Given('I create a premises with all necessary details', () => {
 
   page.getLocalAuthorityAreaIdByLabel('North Lanarkshire', 'localAuthorityAreaId')
 
-  cy.get('@localAuthorityAreaId').then(localAuthorityAreaId => {
+  page.getCharacteristicIdByLabel('Park nearby', 'parkNearbyCharacteristicId')
+  page.getCharacteristicIdByLabel('Floor level access', 'floorLevelAccessCharacteristicId')
+
+  cy.then(function _() {
```
Comment (richpjames) [inline e2e/tests/stepDefinitions/manage.ts:35]: Why the underscore? keeping the linter happy?

**Review by richpjames — DISMISSED** [review]
LGTM. 
Really well structured and described PR - thanks Bryan!

File: e2e/tests/stepDefinitions/manage.ts (hunk 2 above)
Comment (bryangaledxw) [reply e2e/tests/stepDefinitions/manage.ts:35]: Basically 😭

Can't use arrow notation here, as then you lose access to the function's `this` object, which I'm using to get the aliased values in a tidier way than lots of nested `cy.get('@somealias').then()` calls. And there's the convention of using `_` as a prefix for function param names that are unused but needed to satisfy a signature, or for unused variables when destructuring a tuple

File: server/utils/characteristicUtils.ts (hunk 1 above)
Comment (bryangaledxw) [reply server/utils/characteristicUtils.ts]: Yeah, I don't like it but best I could come up with was that or filterAndSortCharacteristics

Comment (bryangaledxw) [reply server/utils/characteristicUtils.ts]: Actually, typing that out, maybe that's fine, brb 😛

**Review by richpjames — APPROVED** [review]

**Pull request merged**